composer/loggers/in_memory_logger.py

```diff
@@ -74,3 +75,50 @@ def log_metric(self, timestamp: Timestamp, log_level: LogLevel, data: TLogData):
             self.data[k].append((timestamp, log_level, v))
         self.most_recent_values.update(data)
         self.most_recent_timestamps.update({k: timestamp for k in data})
+
+    def get_timeseries(self, metric: str, time_unit: Union[str, None] = None):
+        """Returns logged data as dict containing values of a desired metric over time.
+
+        Args:
+            metric (str): Metric of interest. Must be present in self.data.keys().
+            time_unit(str or None, optional) Which unit of time to return. Options =
+                {"batch", "epoch", "sample", "token", "duration", None}. If ``None``,
+                will return all available units of time. Default=``None``.
+
+        Returns:
+            timeseries (dict): Dictionary in which one key is ``metric``, and the
+                associated value is a list of values of that metric. The remaining key(s)
+                are each a unit of time, and the associated values are each a list of
```

Comment:
From a typing perspective, I'm not a huge fan of mixing the value types of a Dict.The annotation of the returned dictionary would be something like `Dict[Union[str, TimeUnit], Union[List[TLogData], List[Time]]]`.

I'm wondering whether it would make sense to break this into returning a tuple where one element is the data and the other is a dictionary of the timestamps -- e.g. `Tuple[List[TLogData], Dict[TimeUnit, List[Time]]`?

Reply:
The return type here should always be `Dict[str, TLogData]`. Even for time units, the code below does:
```
timeseries[time_unit].append(getattr(timestamp, time_unit).value)
```

where `time_unit` is a string, and the latter is compatible with `TLogData`.